objects/src/accounts/delta/builder.rs

```diff
@@ -0,0 +1,54 @@
+use alloc::vec::Vec;
+
+use super::{AccountStorageDelta, StorageMapDelta, Word};
+
+#[derive(Clone, Debug, Default)]
+pub struct AccountStorageDeltaBuilder {
+    pub cleared_items: Vec<u8>,
+    pub updated_items: Vec<(u8, Word)>,
+    pub updated_maps: Vec<(u8, StorageMapDelta)>,
+}
+
+impl AccountStorageDeltaBuilder {
+    // CONSTRUCTORS
+    // ===========================================================================================
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    // MODIFIERS
+    // ===========================================================================================
+    pub fn add_cleared_items<I>(mut self, items: I) -> Self
+    where
+        I: IntoIterator<Item = u8>,
+    {
+        self.cleared_items.extend(items);
+        self
+    }
+
+    pub fn add_updated_items<I>(mut self, items: I) -> Self
+    where
+        I: IntoIterator<Item = (u8, Word)>,
+    {
+        self.updated_items.extend(items);
+        self
+    }
+
+    pub fn add_updated_maps<I>(mut self, items: I) -> Self
+    where
+        I: IntoIterator<Item = (u8, StorageMapDelta)>,
+    {
+        self.updated_maps.extend(items);
+        self
+    }
+
+    // BUILDERS
+    // ===========================================================================================
+    pub fn build(self) -> AccountStorageDelta {
+        AccountStorageDelta {
+            cleared_items: self.cleared_items,
+            updated_items: self.updated_items,
+            updated_maps: self.updated_maps,
+        }
```

Comment:
We should probably validate the struct before returning (e.g., by calling `AccountStorageDelta::validate()`). This way, we'll error out in case the delta is constructed incorrectly.

Reply:
added